internal/olm/client/client.go

```diff
@@ -204,6 +206,46 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 				})
 				return false, nil
 			}
+			// Check deployment status
+			if ctx.Err().Error() == "DeadlineExceeded" {
+				dep := &appsv1.Deployment{}
+				depName := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Name
+				depKey := types.NamespacedName{
+					Namespace: key.Namespace,
+					Name:      depName,
+				}
+				if depErr := c.KubeClient.Get(ctx, depKey, dep); depErr != nil {
+					return false, fmt.Errorf("failed to run operator: Error getting deployment: %v", depErr)
+				}
+				for _, s := range dep.Status.Conditions {
+					if s.Type == "Available" && s.Status == "True" {
+						return false, fmt.Errorf("failed to run operator: Deployment is available, but CSV pending: %v", err)
+					}
+					if s.Type == "Progressing" || s.Type == "ReplicaFailure" {
+						// Dig deep and return  error message.
+						podList := &coreV1.PodList{}
+						podKey := types.NamespacedName{
+							Namespace: key.Namespace,
+						}
+
+						if podErr := c.KubeClient.Get(ctx, podKey, podList); podErr != nil {
+							return false, fmt.Errorf("failed to run operator: Error getting Pods: %v", podErr)
+						}
+						// Loop through pods, and respective containers.
+						for _, p := range podList.Items {
+							for _, c := range p.Status.Conditions {
+								if c.Type == "ContainersReady" && c.Status == "False" {
+									for _, cs := range p.Status.ContainerStatuses {
+										if !cs.Ready {
+											return false, fmt.Errorf("failed to run operator: error while pod coming up: %v", cs.State.Waiting.Message)
+										}
+									}
+								}
+							}
```

Comment:
Just poll the [pod's phase](https://godoc.org/k8s.io/api/core/v1#PodPhase) for `Running` or `Succeeded`.

Reply:
When I did a quick test, The container statuses contained the actual error, even when `pod.phase == Running`. So, need to check the container statuses to report the actual error.